### PR TITLE
Added method to factorize using sieve of eratosthenes

### PR DESCRIPTION
**Pull request by aarush22**
Added another method to factor a number, using sieve of eratosthenes with a little trick i learned here http://codeforces.com/blog/entry/7262.

### CODE REVIEW

**Comment by choyiny** [comment]
The variable names need more information, instead of `v` put what it really does.
Also, can you convert it to use BigIntegers Class? #3

**Comment by aarush22** [comment]
I have updated the variable names
This method is based on sieve of eratosthenes, which uses array indexing to mark all the non-primes thus a BigInteger Implementation would be redundant since the maximum size of an array possible(heap size) is around 10^7. The benefit of this algo is its exceptionally small **time complexity**.

File: src/main/java/in/choy/math/ia/Main.java [hunk 1]
```diff
@@ -255,12 +255,63 @@ public static void pollardsRhoMethod(int num) {
...
+		
+		
+		/*	This method uses the sieve of eratosthenes with 
+		 *  a little trick i learned from here :-
+		 *  http://codeforces.com/blog/entry/7262 
+		 *  The limitation for this method is that the biggest
+		 * 	integer can only be the size of 10^7
+		 * 	i.e. the maximum allocatable size to an array/list   */
+		
+		final long bfStartTime = System.currentTimeMillis();
```
Comment (choyiny) [inline src/main/java/in/choy/math/ia/Main.java]: name this esStartTime

File: src/main/java/in/choy/math/ia/Main.java [hunk 2]
```diff
@@ -255,12 +255,63 @@ public static void pollardsRhoMethod(int num) {
...
+		{
+			int i=smallestPrime[num];
+			System.out.println("SF: " + i + " is a factor");
+			System.out.println("SF: " + num / i + " is another factor");
+		}
+		else
+		{
+			System.out.println("SF: The number is a prime");
+		}
+		final long bfEndTime = System.currentTimeMillis();
```
Comment (choyiny) [inline src/main/java/in/choy/math/ia/Main.java]: esEndTime

File: src/main/java/in/choy/math/ia/Main.java [hunk 3]
```diff
@@ -255,12 +255,63 @@ public static void pollardsRhoMethod(int num) {
...
+		 * 	integer can only be the size of 10^7
+		 * 	i.e. the maximum allocatable size to an array/list   */
+		
+		final long bfStartTime = System.currentTimeMillis();
+		int []smallestPrime=new int[num];
+		bfLooped=Sieve(num,smallestPrime);
+		if(num!=smallestPrime[num])
+		{
+			int i=smallestPrime[num];
+			System.out.println("SF: " + i + " is a factor");
```
Comment (choyiny) [inline src/main/java/in/choy/math/ia/Main.java]: ES:

**Comment by aarush22** [comment]
any other changes you want to highlight ?

**Comment by choyiny** [comment]
Thanks, Merged!

**Pull request merged**